### ISSUE TITLE
zipcode matcher

### ISSUE THREAD

**Issue by chendaniely** [bi-sdal/sdalr#6]
@ZarniHtet13

I can work with you on putting it in the package, just need to know what the 'final' function code is

**Comment by inrazhtet** [comment]
@chendaniely , I will pull together the latest one, I think I have and break it down a bit to make it look nicer and will talk with you tmr on this. Sounds good?
Thanks!

**Comment by chendaniely** [comment]
you can also paste whatever you have in this issue so I can run it and see how it works.

**Comment by inrazhtet** [comment]
I have the source code under: 
ari_base_char/src/zarni90/zip_matcher.RMD

It has a rudimentary example. We can probably improve the code by writing a helper function for processing zip codes. Thanks!

` 
---
title: "zip_map_func"
author: "Zarni Htet"
date: "6/5/2017"
output: html_document
---

```{r}
#########
#This function expects a ACS table code and the zipcode list you want to match it to
#########
library(acs)
library(dplyr)
library(stringr)
zip_matcher <- function(tablecode, ziplist){
    #Get everything in zip code
    FX.VA<-geo.make(zip.code = "*")
    #Fetch the particular table
    FX.VA.TP<-acs.fetch(geography=FX.VA, endyear=2015, table.number=tablecode, col.name="pretty")
    #May need to modify depending on table
    FXCStats<-data.frame(estimate(FX.VA.TP), round(standard.error(FX.VA.TP),0))
    #I have deleted out the colnames. We can put them up when we need to.
    #colnames(FXCStats)<-c("Total","SE.Total")
    #Establishing Zipcode from Rownames
    FXCStats$zipcode <- row.names(FXCStats)
    FXCStats$zipcode
    #Spliting the String part and the numeric part of Zipcode
    zip_split <- str_split(FXCStats$zipcode, pattern = " ", n = 2)
    zipcode <- sapply(zip_split, function(x) x[2])
    FXCStats$zipcode <- zipcode
    #Importing the Master ZipCode File
    ziplist <- as.data.frame(ziplist)
    colnames(ziplist) <- "zipcode"
    ziplist$zipcode <- as.character(ziplist$zipcode)
    FXCStats$zipcode <- as.character(FXCStats$zipcode)
    #Joining the two tables by ZipCode
    FX_Zip_Final <- ziplist %>% left_join(FXCStats, by = "zipcode")
    return (FX_Zip_Final)
}

####################
###Some examples
####################
#Population by Age Group and Sex
#Getting the zipcodes of Fairfax county
ziplist <- rio::import("~/git/comm_fairfax/data/fairfaxZipcodes.txt")
pop_age_group <- zip_matcher("B01003", ziplist)
#write.csv(pop_age_group, file="~/git/comm_fairfax/data/comm_fairfax/working/Pop_Age_Group.csv")

#Population by Race/Ethnicity
pop_race_ethnicity <- zip_matcher("B03002", ziplist)
View(pop_race_ethnicity)
#write.csv(pop_race_ethnicity, file = "~/git/comm_fairfax/data/comm_fairfax/working/Pop_Race_Ethnicity.csv")

```

Sample
```{r}
 FX.VA<-geo.make(zip.code = "*")
 #Fetch the particular table
 FX.VA.TP<-acs.fetch(geography=FX.VA, endyear=2015, table.number=, col.name="pretty")
```



`

**Comment by inrazhtet** [comment]
@chendaniely , could you share me the format that I should abide by again, please?
I am going to work on this now. Thanks!

**Comment by inrazhtet** [comment]
@chendaniely , hey, I have wrapped the zip_matcher. It's under comm_fairfax/src/zarni90/zip_matcher.Rmd.

Thanks!
Zarni

**Comment by inrazhtet** [comment]
```r
#title: "zip_matcher"
#author: "Zarni"
#date: "6/28/2017"
#output: html_document

##########################################################################################
#This set of functions take in a list of ACS table code and a list of zip codes and get all permutation of data set for each of the column. For now, the function to run with a list of table code is still under construction.
##########################################################################################

#Loading the needed data packages
library(acs)
library(dplyr)
library(stringr)

##########################################################################################
#Zip_Matcher: function takes in a single table code, a list of zipcodes (must be put in as character) and the year the ACS survey is taken
##########################################################################################

zip_matcher <- function(tablecode, ziplist, endyear){
    #Get data for all the zipcodes
    US<-geo.make(zip.code = "*")
    #Fetch the particular table
    US.TP<-acs.fetch(geography=US, endyear=endyear, table.number=tablecode, col.name="pretty")
    #Pulling out the Stats we want
    USStats<-data.frame(estimate(US.TP), round(standard.error(US.TP),0))
    #Establishing Zipcode from Rownames
    USStats$zipcode <- row.names(USStats)
    #Spliting the String part and the numeric part of Zipcode
    USStats$zipcode <- zip_acs_processing(USStats$zipcode)
    #Importing and Processing the Master ZipCode File
    #For leading zero zip codes, you should have it formatted as character to begin with.
    ziplist <- input_zip_processing(ziplist)
    ziplist$zipcode <- as.character(ziplist$zipcode)
    USStats$zipcode <- as.character(USStats$zipcode)
    #Joining the two tables by ZipCode
    US_Zip_Final <- ziplist %>% left_join(USStats, by = "zipcode")
    return (US_Zip_Final)
}

##########################################################################################
#zip_acs_processing: this helper function cleans the zipcodes from the ACS table. It takes in
#                    a list of zipcodes
##########################################################################################
zip_acs_processing <- function(zipcode){
    zip_split <- str_split(zipcode, pattern = " ", n = 2)
    zipcode <- sapply(zip_split, function(x) x[2])
    return(zipcode)
}

##########################################################################################
#input_zip_processing: this helper function cleans the zipcodes that the user provides and
#                      formats it as well as takes care of edge cases like leading "00"
##########################################################################################
input_zip_processing <- function(zipcode){
    zipcode <- as.data.frame(zipcode)
    colnames(zipcode) <- "zipcode"
    return (zipcode)
}

####################
###Some examples
####################
#Population by Age Group and Sex
#Getting the zipcodes of Fairfax county
ziplist <- rio::import("~/git/comm_fairfax/data/fairfaxZipcodes.txt")
pop_age_group_2015 <- zip_matcher("B01003", ziplist, 2015)
pop_age_group_2014 <- zip_matcher("B01003", ziplist, 2014)

#write.csv(pop_age_group, file="~/git/comm_fairfax/data/comm_fairfax/working/Pop_Age_Group.csv")

#Population by Race/Ethnicity
pop_race_ethnicity <- zip_matcher("B03002", ziplist)
View(pop_race_ethnicity)
#write.csv(pop_race_ethnicity, file = "~/git/comm_fairfax/data/comm_fairfax/working/Pop_Race_Ethnicity.csv")


FX.VA<-geo.make(zip.code = "*")
#Fetch the particular table
FX.VA.TP<-acs.fetch(geography=FX.VA, endyear=2015, table.number=, col.name="pretty")

```

**Comment by chendaniely** [comment]
fixed #7
